### PR TITLE
Infrastructure: bump vcpkg hash

### PR DESCRIPTION
**Pull request by lebdron**
### Description of the Change
Update dependencies

### Benefits
arm64 build

### Possible Drawbacks 
None

### CODE REVIEW

File: vcpkg/VCPKG_COMMIT_SHA [hunk 1]
```diff
@@ -1 +1 @@
-f3db66b403840b24ea2612d09cca30a5285f5ea3
+9ab3bafca4de5dec5a5a8b60f1d654cae7ad95aa
```
Comment (baziorek) [inline vcpkg/VCPKG_COMMIT_SHA:1]: The only drawback in the new version of VCPKG is that it contains new version of pglibrary, which needs newer `cmake` than it is installable normally with `sudo apt-get install cmake`. ubuntu server 20.10 has version: 3.16.3, but pglibrary wants:  3.17. Solution is to install `cmake` with `sudo snap install cmake --classic`.

I think I'll ignore that fact in my documentation, soon version ubuntu 21.04:).

File: vcpkg/build_iroha_deps.sh [hunk 2]
```diff
@@ -8,5 +8,12 @@ git clone https://github.com/microsoft/vcpkg $vcpkg_path
 git -C $vcpkg_path checkout $(cat "$iroha_vcpkg_path"/VCPKG_COMMIT_SHA)
 for i in "$iroha_vcpkg_path"/patches/*.patch; do git -C $vcpkg_path apply --ignore-whitespace $i; done;
 $vcpkg_path/bootstrap-vcpkg.sh
+
+ARCH="$(uname -m)"
+
+if [ "$ARCH" = "armv7l" -o "$ARCH" = "aarch64" -o "$ARCH" = "s390x" -o "$ARCH" = "ppc64le" ]; then
+    export VCPKG_FORCE_SYSTEM_BINARIES=1
```
Comment (baziorek) [inline vcpkg/build_iroha_deps.sh:15]: it is impacting commands in the script, but calling (according to documentation:
https://iroha.readthedocs.io/en/master/build/index.html#linux-and-macos)
requires the flag again:
```
VCPKG_FORCE_SYSTEM_BINARIES=1 vcpkg/vcpkg integrate install
```
I'll describe its in documentation:)

**Review by baziorek — APPROVED** [review]
I've checked on PC and RPi 4 - it is working automatically. There is only problem with cmake for pglib (as described in comment), but it is not a problem for programmers.

**Review by iceseer — APPROVED** [review]

File: irohad/consensus/yac/impl/peer_orderer_impl.cpp [hunk 3]
```diff
@@ -5,6 +5,7 @@
 
 #include "consensus/yac/impl/peer_orderer_impl.hpp"
 
+#include <algorithm>
```
Comment (kuvaldini) [inline irohad/consensus/yac/impl/peer_orderer_impl.cpp:8]: really need?

File: libs/logger/logger.hpp [hunk 4]
```diff
@@ -12,20 +12,34 @@
...
 // struct, leading to compilation issues
 #undef interface
 
+namespace fmt {
+  /// Allows to log objects, which have toString() method without calling it,
+  /// e.g. log.info("{}", myObject)
+  template <typename T>
+  struct formatter<
+      T,
+      std::enable_if_t<std::is_same<decltype(std::declval<T>().toString()),
```
Comment (kuvaldini) [inline libs/logger/logger.hpp:27]: toString() may also return const&, or even string_view. Let's use std::decay<>
```
std::is_same_v<std::decay_t<decltype(std::declval<T>().toString())>,std::string>
```

File: test/framework/test_logger.hpp [hunk 5]
```diff
@@ -9,6 +9,14 @@
 #include "logger/logger.hpp"
 #include "logger/logger_manager_fwd.hpp"
 
+/// Allows to log objects, which have toString() method without calling it, e.g.
+/// log.info("{}", myObject)
```
Comment (kuvaldini) [inline test/framework/test_logger.hpp:13]: /// std::cerr << myObject << std::endl;

**Review by kuvaldini — CHANGES_REQUESTED** [review]

File: irohad/consensus/yac/impl/peer_orderer_impl.cpp (hunk 3 above)
Comment (lebdron) [reply irohad/consensus/yac/impl/peer_orderer_impl.cpp:8]: https://github.com/hyperledger/iroha/blob/72ec57c55b9b65b19a459ec54038014109b553a6/irohad/consensus/yac/impl/peer_orderer_impl.cpp#L29
https://en.cppreference.com/w/cpp/algorithm/random_shuffle
> Defined in header `<algorithm>`

File: libs/logger/logger.hpp (hunk 4 above)
Comment (baziorek) [reply libs/logger/logger.hpp:27]: I've checked and changing the code above into:
```
  template <typename T>
  struct formatter<
      T,
      std::enable_if_t<std::is_same<std::decay_t<decltype(std::declval<T>().toString())>,
```
does not add any compilation error (I've checked with tests).

**Review by kuvaldini — APPROVED** [review]

**Comment by kuvaldini** [comment]
### Previous jenkins before merge to trunck has failed. Restarted. Please check @lebdron

**Comment by lebdron** [comment]
`"boost::filesystem::status: Function not implemented: "/tmp""` seems to be CI issue as tests pass manually in the same Docker container.

**Pull request merged**